api/service.go

```diff
@@ -110,6 +336,83 @@ func (s *service) Listen(req *pb.ListenRequest, server pb.API_ListenServer) erro
 	return nil
 }
 
+func (s *service) processCreateRequest(req *pb.ModelCreateRequest, createFunc func(...interface{}) error) (*pb.ModelCreateReply, error) {
```

Comment:
These "process..." methods are to consolidate code that is used in both the contexts of implicit transaction calls and explicit transactions. The function to call to actually execute the call is passed in (`model.Create` vs `txn.Create`, for example)